src/main/java/hudson/plugins/collabnet/util/Helper.java

```diff
@@ -65,30 +73,37 @@ public static String getToken(URL ctfUrl, String ctfUserName, String ctfPassword
         return token;
     }
 
-    public static String getWebhookToken(String webhookUrl, String webhookUsername, String webhookPassword,
-                                         TaskListener listener)
-            throws
-            IOException {
+    public static String getWebhookUrl(String ctfUrl) throws IOException {
+        String webhookUrl = null;
         CloseableHttpClient client = null;
         CloseableHttpResponse response = null;
-        String token = null;
+        String webhookEndPoint = null;
         try {
-            client = CNFormFieldValidator.getHttpClient();
-            HttpPost httpPost = new HttpPost(webhookUrl);
-            StringEntity entity = new StringEntity(getLoginData(webhookUsername, webhookPassword).toString());
-            httpPost.setEntity(entity);
-            httpPost.setHeader("Accept", "application/json");
-            httpPost.setHeader("Content-type", "application/json");
-            response = client.execute(httpPost);
-            if(response.getStatusLine().getStatusCode() == 200) {
-                token = JSONObject.fromObject(EntityUtils.toString(response.getEntity())).getJSONObject("Response").
-                        get("SessionToken").toString();
+            webhookUrl =  ctfUrl.concat(":3000/publishers/v4");
+            client = HttpClients.createDefault();
+            HttpGet httpGet = new HttpGet(webhookUrl);
+            URI uri = new URIBuilder(httpGet.getURI()).addParameter("q",
+                    queryParam).build();
+            httpGet.setURI(uri);
+            response = client.execute(httpGet);
+            String responseMsg = EntityUtils.toString(response.getEntity());
+            if(!responseMsg.isEmpty()) {
+                JSONObject responseObject = JSONObject.fromObject(responseMsg);
+                int responseCode = Integer.parseInt(responseObject.get("HTTPStatusCode").toString());
+                if (responseCode == 200) {
+                    JSONArray jsonArray = (JSONArray) responseObject.get("Response");
+                    if (!jsonArray.isEmpty()) {
+                        JSONObject jsonObject = (JSONObject) jsonArray.get(0);
+                        String endpointUrl = jsonObject.get("WebhookEndpoint").toString();
+                        webhookEndPoint = endpointUrl.replace("<<EventName>>", eventName);
+                    }
+                } else {
+                    logger.log(Level.INFO,"Error in getting the webhook endpoint.");
+                    logger.log(Level.INFO,responseObject.get("ErrorText").toString());
+                }
             }
-        }catch (IOException e) {
-            String errMsg = e.getMessage();
-            logMsg(errMsg, listener, logger, e);
         } catch (Exception e) {
-            logger.log(Level.INFO,"TeamForge Associations - " + e.getLocalizedMessage(), e);
+            e.printStackTrace();
```

Comment:
Please log exceptions instead of 'printStackTrace'

Reply:
Logged exceptions as earlier.